Tests/Microsoft.AppCenter.Crashes.Test.Windows/HandledErrorTest.cs

```diff
@@ -0,0 +1,237 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.AppCenter.Channel;
+using Microsoft.AppCenter.Crashes.Ingestion.Models;
+using Microsoft.AppCenter.Ingestion.Http;
+using Microsoft.AppCenter.Ingestion.Models;
+using Microsoft.AppCenter.Ingestion.Models.Serialization;
+using Microsoft.AppCenter.Storage;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+using Newtonsoft.Json;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+
+namespace Microsoft.AppCenter.Crashes.Test.Windows
+{
+    [TestClass]
+    public class HandledErrorTest
+    {
+        private static IHttpNetworkAdapter StartCrashes()
+        {
+            var mockHttpNetworkAdapter = Mock.Of<IHttpNetworkAdapter>();
+            var storage = new Storage.Storage(new StorageAdapter("test.db"));
+            var ingestion = new IngestionHttp(mockHttpNetworkAdapter);
+            var channelGroup = new ChannelGroup(ingestion, storage, "app secret");
+            Crashes.Instance = new Crashes();
+            Crashes.SetEnabledAsync(true).Wait();
+            Crashes.Instance.OnChannelGroupReady(channelGroup, "app secret");
+            return mockHttpNetworkAdapter;
+        }
+
+        [TestMethod]
+        public void TrackErrorWithValidParameters()
+        {
+            var mockHttpNetworkAdapter = StartCrashes();
+            var semaphore = new SemaphoreSlim(0);
+            HandledErrorLog actualLog = null;
+            Mock.Get(mockHttpNetworkAdapter).Setup(adapter => adapter.SendAsync(
```

Comment:
This is the exact same for several tests, can we do the setup in the initialization or move it to a helper method?

Reply:
I personally think that the test gets a bit obfuscated the more we move into the setup. Especially with mocking

Reply:
Resolving as this is also a bit more of an opinion type thing.

Reply:
We can't, try it ;) You will get an error that an out variable cannot be set from a lambda expression.